To: vim_dev@googlegroups.com
Subject: Patch 7.3.236
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.236 (after 7.3.232)
Problem:    Python 3 doesn't compile without +multi_byte 
Solution:   Use "latin1" when MULTI_BYTE is not defined. (lilydjwg)
Files:      src/if_python3.c


*** ../vim-7.3.235/src/if_python3.c	2011-06-19 00:27:46.000000000 +0200
--- src/if_python3.c	2011-06-26 19:10:57.000000000 +0200
***************
*** 70,76 ****
  
  #define PyInt Py_ssize_t
  #define PyString_Check(obj) PyUnicode_Check(obj)
! #define PyString_AsBytes(obj) PyUnicode_AsEncodedString(obj, (char *)p_enc, NULL);
  #define PyString_FreeBytes(obj) Py_XDECREF(bytes)
  #define PyString_AsString(obj) PyBytes_AsString(obj)
  #define PyString_Size(obj) PyBytes_GET_SIZE(bytes)
--- 70,76 ----
  
  #define PyInt Py_ssize_t
  #define PyString_Check(obj) PyUnicode_Check(obj)
! #define PyString_AsBytes(obj) PyUnicode_AsEncodedString(obj, (char *)ENC_OPT, NULL);
  #define PyString_FreeBytes(obj) Py_XDECREF(bytes)
  #define PyString_AsString(obj) PyBytes_AsString(obj)
  #define PyString_Size(obj) PyBytes_GET_SIZE(bytes)
***************
*** 661,667 ****
  
      /* PyRun_SimpleString expects a UTF-8 string. Wrong encoding may cause
       * SyntaxError (unicode error). */
!     cmdstr = PyUnicode_Decode(cmd, strlen(cmd), (char *)p_enc, NULL);
      cmdbytes = PyUnicode_AsEncodedString(cmdstr, "utf-8", NULL);
      Py_XDECREF(cmdstr);
      PyRun_SimpleString(PyBytes_AsString(cmdbytes));
--- 661,667 ----
  
      /* PyRun_SimpleString expects a UTF-8 string. Wrong encoding may cause
       * SyntaxError (unicode error). */
!     cmdstr = PyUnicode_Decode(cmd, strlen(cmd), (char *)ENC_OPT, NULL);
      cmdbytes = PyUnicode_AsEncodedString(cmdstr, "utf-8", NULL);
      Py_XDECREF(cmdstr);
      PyRun_SimpleString(PyBytes_AsString(cmdbytes));
***************
*** 1463,1469 ****
      }
      *p = '\0';
  
!     result = PyUnicode_Decode(tmp, len, (char *)p_enc, NULL);
  
      vim_free(tmp);
      return result;
--- 1463,1469 ----
      }
      *p = '\0';
  
!     result = PyUnicode_Decode(tmp, len, (char *)ENC_OPT, NULL);
  
      vim_free(tmp);
      return result;
*** ../vim-7.3.235/src/version.c	2011-06-26 05:36:07.000000000 +0200
--- src/version.c	2011-06-26 19:12:12.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     236,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
228. You spend Saturday night making the counter on your home page
     pass that 2000 mark.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
